### DISCUSSION TITLE
Renewal, Healthstone, Healthpot

### DISCUSSION THREAD

**Discussion by laxzzi** [HgAlexx/Buffet#22] [Q&A]
Hello there!

I see how I could create a custom macro that can use items, but is it possible to put a spell in there?

I want to make the following happen in this prio:

1. Renewal (druid talent, spellID 108238, 90 sec CD for castsequence) - If possible, only show if talented? Not required.
2. Healthstone if available
3. Spiritual Healing Potion

No difference in combat/non-combat, dont want to use any modifiers. Just a one-button-macro for all of the above.

I would be so happy if any of this was possible 😍

- Wixii

**Comment by HgAlexx** [comment]
Hello,

Try this:

```
local bests, cache = ...

local content = {}

local usable, nomana = IsUsableSpell("Renewal")
if usable and not nomana then
   table.insert(content, "reset=90 Renewal")
end

-- check for healthstone
if bests.healthstone then
    table.insert(content, "item:" .. bests.healthstone)
end

-- check for Spiritual Healing Potion
if cache and cache[171267] then
    table.insert(content, "item:171267")
end

return "#showtooltip\n/castsequence " .. table.concat(content, ",")
```

I don't know if the reset will work properly, and also note that the check on healthstone will only work out of combat.
Once used in combat, it will still show up in the sequence until you get out of combat.

**Reply by laxzzi** [reply]
Hello! Thanks a lot for you reply!

I havent gotten ahold of a warlock to test the Healthstone yet, but the castsequence doesnt seem to reset while in combat :(

**Reply by HgAlexx** [reply]
The reset count from the last time you used the sequence, not from the first spell used.

**Reply by laxzzi** [reply]
Aah, hmm.. Then I think the exact thing I am after is not possible. But this brought me closer! Thank you for the quick replies! :)

**Reply by HgAlexx** [reply]
As I understand it, Blizzard doesn't want you to be able to make a macro which cycle through spells based on cooldown so you can't have a one click do it all :(
